filebeat/harvester/reader/docker_json.go

```diff
@@ -39,27 +95,19 @@ func (p *DockerJSON) Next() (Message, error) {
 			return message, err
 		}
 
-		var line dockerLog
-		dec := json.NewDecoder(bytes.NewReader(message.Content))
-		if err = dec.Decode(&line); err != nil {
-			return message, errors.Wrap(err, "decoding docker JSON")
-		}
+		var dockerLine dockerLog
+		var crioLine crioLog
 
-		if p.stream != "all" && p.stream != line.Stream {
-			continue
+		if strings.HasPrefix(string(message.Content), "{") {
```

Comment:
I wonder if we should make this a config option instead of "guessing" it here? Are there potentially other formats which will be added?

Reply:
I don't think new formats will be added, CRI is the standard Kubernetes went for, the idea of this is to support both the standard and old Docker format.
I'm ok with autodetecting it as of today, we can change this in the future if we see the point